internal/version/envoy.txt

```diff
@@ -1 +1 @@
-1.17.1
\ No newline at end of file
+1.18.3
```

Comment:
the tests for site/** caught this btw

Reply:
do we need now? why not just read the latest version from envoy_versions.json? (I guess you plan to do in a separate PR).

Reply:
renamed and reworded docs around "last known envoy". This was indeed an issue of confusion, so thanks for thinking about it!